langchain/src/callbacks/base.ts

```diff
@@ -187,6 +188,40 @@ abstract class BaseCallbackHandlerMethodsClass {
     parentRunId?: string,
     tags?: string[]
   ): Promise<void> | void;
+
+  /**
+   * Called when embedding throws an error, after call to embedding.
+   * with the error and the run ID
+   */
+  handleEmbeddingError?(
+    err: Error,
+    runId: string,
+    parentRunId?: string
+  ): Promise<void> | void;
+
+  /**
+   * Called when embedding is started, before call to embedding.
+   * with the input text and the run ID
+   */
+  handleEmbeddingStart?(
+    embedding: Serialized,
+    texts: string[],
+    runId: string,
+    parentRunId?: string,
+    extraParams?: Record<string, unknown>,
+    tags?: string[],
+    metadata?: Record<string, unknown>
+  ): Promise<void> | void;
+
+  /**
+   * Called after embedding is completed, before it exits.
+   * with embeddings vector and the run ID
+   */
+  handleEmbeddingEnd?(
+    vector: number[],
```

Comment:
If `start` accepts multiple strings, the `end` handler should accept a list of vectors I'd assume. @agola11 think we should do the runID looping outside the handler though, right?

Reply:
The function handleEmbeddingStart is designed to process multiple documents during a single embedDocuments call. However, each subsequent result from this process is individually fed into handleEmbeddingEnd calls. This structure was implemented to mirror the behavior of handleLLMStart and handleLLMEnd. In those functions, handleLLMStart deals with several prompts simultaneously, but handleLLMEnd processes each resulting output one at a time. This consistency in design was suggested by @nfcampos.

Reply:
Right! Yes yes thanks for clarifying